### PR TITLE
Initial commit: Version 0.0.1 - Server Functionality Testing

### PR DESCRIPTION
**Pull request by A-FlorezMeyer**
This commit marks the initial release, version 0.0.1, of our server application. This version primarily focuses on testing the server's core functionality and establishing the foundational structure of the project. It includes the initial setup of the project's architecture and the implementation of the first three message handlers. These components are crucial for processing incoming data and represent the first step towards a fully functional server. This version is intended for internal testing and validation of the server's basic operations. Future updates will build upon this foundation to introduce additional features and optimizations.

### CODE REVIEW

File: src/models/Device.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Device {
+    constructor(id) {
+        this.id = id;
+        // Otros atributos relevantes, como estado de la batería, paso, etc.
```
Comment (B4rz99) [inline src/models/Device.js]: Other relevant attributes like battery status, steps, etc.

File: src/models/Device.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    }
+
+    // Métodos para manejar el estado del dispositivo, como actualizar batería, paso, etc.
```
Comment (B4rz99) [inline src/models/Device.js]: Methods to control the device's status as update battery, steps, etc.

File: src/models/LocationData.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    }
+    position(coordinate, direction) {
+        if (direction === 'S' || direction === 'W') {
+            return -coordinate;
+        }
+        return coordinate;
+    }
+    todatetime(date,time){
+        const year = `20${date.slice(4,6)}`; // Asume un año 2000+
```
Comment (B4rz99) [inline src/models/LocationData.js]: Assumes a year greater than 1999

File: src/types/ALMessage.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        console.log(`Terminal Information: ${locData.terminal}`);
+        
+        
+        let content = 'AL';
+
+
+        let lengthInHex = content.length.toString(16);
+        let length = lengthInHex.padStart(4, '0');
+
+        // Formato de respuesta a terminal
```
Comment (B4rz99) [inline src/types/ALMessage.js]: Terminal answer format

File: src/types/LKMessage.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+//Link keep
+//Once link data every 5 minutes
+//if the terminal has not received the reply data, then 5 minutes reconnect.
+class LKMessage {
+    handle(parts, socket) {
+        const manufacturer = parts[0];
+        const id = parts[1];
+        // Procesa aquí el mensaje LK
```
Comment (B4rz99) [inline src/types/LKMessage.js]: Here the LK message is processed.

File: src/types/LKMessage.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        
+        
+        //content of the response
+        let content = 'LK';
+
+
+        let lengthInHex = content.length.toString(16);
+        let length = lengthInHex.padStart(4, '0');
+
+        // Formato de respuesta a terminal
```
Comment (B4rz99) [inline src/types/LKMessage.js]: Terminal answer format

File: src/types/UDMessage.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+//Position data report
+//The terminal reports the position and the condition 
+//information according to the upload time, does not need the platform reply.
+class UDMessage {
+    handle(parts, socket) {
+
+        const data = parts[3];
+        const locData = new LocationData(data);
+        // Procesa aquí el mensaje UD
```
Comment (B4rz99) [inline src/types/UDMessage.js]: Here the UD message is processed.

File: src/types/UDMessage.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        
+        // Imprimir la información procesada en la consola
```
Comment (B4rz99) [inline src/types/UDMessage.js]: Printing the processed information

File: src/types/UDMessage.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        console.log(`Power: ${locData.power}`);
+        console.log(`Steps: ${locData.steps}`);
+        console.log(`Roll: ${locData.roll}`);
+        console.log(`Terminal Information: ${locData.terminal}`);
+        
+        //content of the response
+    
+
+        //POSITION DATA REPORT
+        //no se responde a terminal
```
Comment (B4rz99) [inline src/types/UDMessage.js]: No answer in terminal.

**Review by B4rz99 — CHANGES_REQUESTED** [review]
There are some comments written in spanish, they all need to be in english. Everythingelse seems to be fine.

**Review by A-FlorezMeyer — COMMENTED** [review]
Updated comments

**Review by B4rz99 — APPROVED** [review]
Changes have been done.

**Pull request merged**